pkg/messageprocessors/javascript/javascript.go

```diff
@@ -379,6 +379,11 @@ func (*host) decodeUplink(
 			msg.NormalizedPayloadWarnings = appendValidationErrors(msg.NormalizedPayloadWarnings, normalizedMeasurements)
 		}
 	}
+
+	// Roundtrip the message in order to convert special number values such as NaN and Infinity to their string form.
+	// TODO: Clean up, or fail, the message (https://github.com/TheThingsNetwork/lorawan-stack/issues/6128).
+	msg.DecodedPayload, _ = structpb.NewStruct(msg.DecodedPayload.AsMap())
```

Comment:
This errors only on invalid UTF-8 it seems. Fine then.

Reply:
I'll actually emit the error in order to avoid showing an empty decoded payload.